DatadogRUM/Tests/Instrumentation/WatchdogTerminations/WatchdogTerminationAppStateManagerTests.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import DatadogInternal
+@testable import DatadogRUM
+import TestUtilities
+
+final class WatchdogTerminationAppStateManagerTests: XCTestCase {
+    // swiftlint:disable implicitly_unwrapped_optional
+    var sut: WatchdogTerminationAppStateManager!
+    var dataStore: RUMDataStore!
+    var featureScope: FeatureScopeMock!
+    var core: PassthroughCoreMock!
+    // swiftlint:enable implicitly_unwrapped_optional
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        featureScope = FeatureScopeMock()
+        dataStore = RUMDataStore(featureScope: featureScope)
+        sut = WatchdogTerminationAppStateManager(
+            featureScope: featureScope,
+            processId: .init()
+        )
+        core = PassthroughCoreMock(
+            context: .mockWith(applicationStateHistory: .mockAppInBackground()),
+            messageReceiver: sut
+        )
+    }
+
+    func testAppStart_SetsIsActive() throws {
+        try sut.start()
+
+        let isActiveExpectation = expectation(description: "isActive is set to true")
+
+        // app state changes
+        core.context.applicationStateHistory.append(.init(state: .active, date: .init()))
+        dataStore.value(forKey: .watchdogAppStateKey) { (appState: WatchdogTerminationAppState?) in
+            XCTAssertTrue(appState?.isActive == true)
+            isActiveExpectation.fulfill()
+        }
+        wait(for: [isActiveExpectation], timeout: 1)
```

Comment:
suggestion/ 🏕️ `FeatureScopeMock` is single-threaded just like `DataStore` it instantiates so I wonder if we can simplify these tests by removing `expectation`. By adding generic extension (in tests) we could reach this syntax:
```swift
XCTAssertEqual((featureScope.rumDataStore.value(forKey: .watchdogAppStateKey) as? WatchdogTerminationAppState)?.isActive, false)
```
and this can benefit other tests we add around `FeatureScopeMock` in the future.

Reply:
I like it